cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/PredefinedNoConfidenceDRep.hs

```diff
@@ -0,0 +1,457 @@
+  void $ H.execCli' execConfig $
+    [ "conway", "governance", "action", "update-committee"
+    , "--testnet"
+    , "--governance-action-deposit", show @Integer minDRepDeposit
+    , "--deposit-return-stake-verification-key-file", stakeVkeyFp
+    ] ++ concatMap (\(prevGovernanceActionTxId, prevGovernanceActionIndex) ->
+                      [ "--prev-governance-action-tx-id", prevGovernanceActionTxId
+                      , "--prev-governance-action-index", show prevGovernanceActionIndex
+                      ]) prevGovActionInfo ++
+    [ "--anchor-url", "https://tinyurl.com/3wrwb2as"
+    , "--anchor-data-hash", proposalAnchorDataHash
+    ] ++ concatMap (\keyHash ->
+                    [ "--add-cc-cold-verification-key-hash", keyHash
+                    , "--epoch", show (100 :: Int)
+                    ]) coldKeyHashes ++
+    [ "--threshold", "0"
+    , "--out-file", proposalFile
+    ]
+
+  proposalBody <- H.note $ baseDir </> "tx.body"
+  txIn <- findLargestUtxoForPaymentKey epochStateView sbe wallet
+
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "build"
+    , "--change-address", Text.unpack $ paymentKeyInfoAddr wallet
+    , "--tx-in", Text.unpack $ renderTxIn txIn
+    , "--proposal-file", proposalFile
+    , "--out-file", proposalBody
+    ]
+
+  signedProposalTx <- signTx execConfig cEra baseDir "signed-proposal"
+                             (File proposalBody) [paymentKeyInfoPair wallet]
+
+  submitTx execConfig cEra signedProposalTx
+
+  governanceActionTxId <- retrieveTransactionId execConfig signedProposalTx
+
+  !propSubmittedResult <- findCondition (maybeExtractGovernanceActionIndex ceo (fromString governanceActionTxId))
+                                        (unFile configurationFile)
+                                        (unFile socketPath)
+                                        (EpochNo 10)
+
+  governanceActionIndex <- case propSubmittedResult of
+                             Left e ->
+                               H.failMessage callStack
+                                 $ "findCondition failed with: " <> displayError e
+                             Right Nothing ->
+                               H.failMessage callStack "Couldn't find proposal."
+                             Right (Just a) -> return a
+
+  return (governanceActionTxId, governanceActionIndex)
+
+delegateToAlwaysNoConfidence
+  :: (MonadTest m, MonadIO m, H.MonadAssertion m, MonadCatch m)
+  => H.ExecConfig
+  -> EpochStateView
+  -> FilePath
+  -> FilePath
+  -> ShelleyBasedEra ConwayEra
+  -> FilePath
+  -> String
+  -> PaymentKeyInfo
+  -> StakingKeyPair
+  -> m ()
+delegateToAlwaysNoConfidence execConfig epochStateView configurationFile socketPath sbe work prefix =
+  delegateToAutomaticDRep execConfig epochStateView configurationFile socketPath sbe work prefix
+                          "--always-no-confidence"
+
+testNoConfidenceProposal
+  :: (MonadTest m, MonadIO m, H.MonadAssertion m, MonadCatch m, Foldable t)
```

Comment:
`HasCallStack`

Reply:
Fixed in: https://github.com/IntersectMBO/cardano-node/pull/5809/commits/62a3cbbad2eaa4573630cc9b7aa40f74f1ae7e41

Reply:
Fixed here: https://github.com/IntersectMBO/cardano-node/pull/5809/commits/62a3cbbad2eaa4573630cc9b7aa40f74f1ae7e41

Reply:
Fixed here: https://github.com/IntersectMBO/cardano-node/commit/35ba8222a2a883d4ad2715ada973affe3c1bd8c7